projects/rocsparse/library/src/common/rocsparse_position_t.cpp

```
@@ -52,112 +59,160 @@ const void* rocsparse::position_t::get_position() const
     return this->m_position;
 }
 
-void rocsparse::position_t::set_max_position_async(hipStream_t stream)
+rocsparse_status rocsparse::position_t::set_position_batch_count(int64_t value, hipStream_t stream)
 {
-    if(this->m_position_indextype == rocsparse_indextype_i32)
+    if((this->m_position != nullptr) && (this->m_batch_count != value))
     {
-        THROW_IF_ROCSPARSE_ERROR(rocsparse::assign_async(
-            (int32_t*)this->m_position, std::numeric_limits<int32_t>::max(), stream));
+        RETURN_IF_HIP_ERROR(rocsparse_hipFreeAsync(this->m_position, stream));
+        this->m_position = nullptr;
     }
-    else
+
+    if(this->m_position == nullptr)
     {
-        THROW_IF_ROCSPARSE_ERROR(rocsparse::assign_async(
-            (int64_t*)this->m_position, std::numeric_limits<int64_t>::max(), stream));
+        RETURN_IF_HIP_ERROR(
+            rocsparse_hipMallocAsync(&this->m_position, sizeof(int64_t) * value, stream));
+        this->m_batch_count = value;
+        std::cout << "set max MISSING " << __LINE__ << std::endl;
+        exit(1);
```

Maybe this is also debug, but I dont like this secret exit. We should return a error status (although it doesnt look like an error?).

---

old school debug, removed